### DISCUSSION TITLE
How to use PicoRV32 arch for Ripes?

### DISCUSSION THREAD

**Discussion by Qpicpicxxz** [mortbopet/Ripes#253] [General]
Hi,
I'm a master student and my work is to add some extensions on PicoRV32, I notice that Ripes has a file which has implemented PicoRV32's cpu core on verilator, and I've built it successfully on my Ubuntu. I wonder if Ripes could be a develop tools for modify cpu cores? Which not just for learning I think, since it's an implementation on a certain RTL code,  and I wonder if it can be a developing golden model? I'm not sure...

**Comment by mortbopet** [comment]
I assume that you've gone through the writeup here (https://github.com/mortbopet/Ripes/blob/master/docs/new_processor_models.md#experimental-verilator-processor-models-in-ripes).

> I wonder if Ripes could be a develop tools for modify cpu cores? Which not just for learning I think, since it's an implementation on a certain RTL code, and I wonder if it can be a developing golden model?

The extent that Ripes (for now) can be used as a development tool is as being an interface for the processor simulation. That is, an interface where you can easily write programs, have them assembled, and then execute them through the Verilator-based simulation, while also being able to inspect the memory contents at any given time.
